### PR TITLE
adds ability to create and generate component specific tokens and Sass

### PR DESCRIPTION
**Pull request by sturobson**
- [x] create build script to generate brand Sass files for Global components.
- [x] create build script to generate Sass files for Brand components.
- [x] Tidy up the outputted Sass files
  - [x] alphabetise the Sass variables for better comprehension.
  - [x] remove Style Dictionary header generator.
  - [x] add date, time, source file, and DO NOT Edit message header.
- [x] add `$tokens--` to referenced Sass variables as SD is not doing it.
- [x] create and generate new `alias` tokens as needed.
- [ ] look at where adding `!default` to Sass variables generated make sense.
  - [ ] implement as needed (wip).
- [x] look at how we can have relative and static sizing or spacing (borders being better in `px` than `em` or `rem` for example.
- [ ] update all the docs
- [x] tidy up possible JS
- [x] is it possible to put the scripts used for global components to be one script?

### CODE REVIEW

**Comment by sturobson** [comment]
To get this working when you're `npm install`ed to the max … run:
```
npm run tokens:components
``` 
in the terminal this script will then:

- clears and deletes all existing `00-tokens`
- generate all literal tokens to create Sass variables from defined Design decisions
- generate the brand specific tokens folder `00-tokens` and `${brand}.tokens.scss` file for global components (only global footer at the moment).
- generate brand component specific `${component}.tokens.scss` file in a folder (`00-tokens`) for each Springer, Nature, and Springer Nature component (none included).

File: toolkits/global/packages/global-forms/scss/00-tokens/_default.tokens.scss [hunk 1]
```diff
@@ -0,0 +1,43 @@
+// Generated on 16/05/2022, 16:40:55
+// Source: design-tokens/componenets/global/global-forms/default.json
+// DO NOT edit directly
+
+$global-forms-checkbox-checked-background-color: $tokens--$background-container-inverted;
```
Comment (alexkilgour) [inline toolkits/global/packages/global-forms/scss/00-tokens/_default.tokens.scss:5]: There seems to be a rogue `$` in these names e.g. `$tokens--$background-container-inverted` should be `$tokens--background-container-inverted`.

Comment (sturobson) [reply toolkits/global/packages/global-forms/scss/00-tokens/_default.tokens.scss:5]: that'll be my bad JS skills at play. I'll take a look

Comment (sturobson) [reply toolkits/global/packages/global-forms/scss/00-tokens/_default.tokens.scss]: resolved in [this commit](https://github.com/springernature/frontend-toolkits/pull/746/commits/c1bb29c2f67085cbb7a9c6a5650dd80e954663e6)

**Comment by sturobson** [comment]
there's a few things here that will have to be created on a separate ticket. 

To further along @Heydon's work on #743 we are merging this work into that and we will pair on updating the relevant changelogs. 

So, in that respect - I'll close this PR as work, up to this point, will become part of #743

**Pull request closed without merging**